configx/provider.go

```diff
@@ -215,7 +218,7 @@ func (p *Provider) validate(k *koanf.Koanf) error {
 // This is unfortunately required due to several limitations / bugs in koanf:
 //
 // - https://github.com/knadh/koanf/issues/77
-// - https://github.com/knadh/koanf/pull/47
+// - https://github.com/knadh/koanf/p	ull/47
```

Comment:
It might be more useful to just support Golang templates here.
Like this: {{ env "GOOGLE_CLIENT_ID" }}

@aeneasr what do you think?

Reply:
Env seems like a good idea. `docker-compose` does something similar by allowing interpolation in the config files. There isn't too much we can do in supporting an array syntax in a generic way since it'll be ambiguous.

How about doing a pass with go templates in `KoanFile.Read()` before running the parser?

Reply:
> How about doing a pass with go templates in `KoanFile.Read()` before running the parser?

Have done just like that.
This is how it looks like in the config file:
{{ .env.GOOGLE_CLIENT_ID }}